### PR TITLE
Workflow to add a checklist for POC issues

### LINKED ISSUE #8028: Introduce a checklist for any Proof of Concepts (POCs) for a feature
Currently, when someone starts on a POC for a given feature, the focus is on coming up with a working solution with a given approach in mind. 

Once the POC has been proved, is when we start working on an overall design for the feature. Often during this design process is when a lot of issues are identified esp. regarding issues like bwc compatibility, etc. 

Introducing a basic checklist for all POCs would help to ensure that these issues are identified beforehand. Even though the expectation is not for the POC to address all of these in its implementation, there should be a plan in place on how to address all of them. 

**Expected behavior**

1. Introduce a new label - "poc" 
2. Add a Github action workflow that adds a comment to all poc issues with a checklist of potential pitfalls to keep in mind during this design.

### PR DESCRIPTION
**Pull request by Poojita-Raj**
### Description
Add a Github action workflow that adds a comment to all issues tagged with label "poc" with a checklist of potential pitfalls to keep in mind during the design of the poc for a feature.

Need a maintainer/code owner to create a label titled "poc" since I am not able to. 

### Related Issues
Resolves #8028 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -97,6 +97,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Add TokenManager Interface ([#7452](https://github.com/opensearch-project/OpenSearch/pull/7452))
 - Add Remote store as a segment replication source ([#7653](https://github.com/opensearch-project/OpenSearch/pull/7653))
 - Add descending order search optimization through reverse segment read. ([#7967](https://github.com/opensearch-project/OpenSearch/pull/7967))
+- Workflow to add a checklist for POC issues ([#8033](https://github.com/opensearch-project/OpenSearch/pull/8033))
```
Comment (kotwanikunal) [inline CHANGELOG.md:100]: I don't think we need a changelog entry for this scenario. The change is for developers and not OpenSearch users.

**Comment by kotwanikunal** [comment]
LGTM. Let's hear out from some more folks before merging it in.

**Review by dblock — APPROVED** [review]
I like it!

**Review by reta — APPROVED** [review]

**Review by owaiskazi19 — APPROVED** [review]
Added a new label `POC`. You might need to check for `POC` instead `poc`.

**Comment by Poojita-Raj** [comment]
> Added a new label `POC`. You might need to check for `POC` instead `poc`.

Label name checks are case-insensitive. This should be good to go. Thanks for creating the label!

**Review by kotwanikunal — APPROVED** [review]

**Pull request merged**